### PR TITLE
fix(lyrics-plus): separate lyrics providers from mode selection

### PR DESCRIPTION
**Pull request by Deskehs**
I messed up git history so I reported the code to latest

continuing from #2330

### CODE REVIEW

File: CustomApps/lyrics-plus/index.js [hunk 1]
```diff
@@ -621,49 +670,58 @@ class LyricsContainer extends react.Component {
...
 			} else if (this.state.genius) {
-				mode = GENIUS;
+				mode = Modes.GENIUS;
 			}
 		}
 
 		const hasNeteaseTranslation = this.state.neteaseTranslation !== null;
-		const language = (this.state.synced || this.state.unsynced) && Utils.detectLanguage(this.state.synced || this.state.unsynced);
+		this.lyricsSource(mode);
+		console.log(this.state.currentLyrics);
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:685]:
```suggestion
```

File: CustomApps/lyrics-plus/index.js [hunk 2]
```diff
@@ -621,49 +670,58 @@ class LyricsContainer extends react.Component {
...
+		const language = () => {
+			if (!this.state.currentLyrics) return;
+			if ([CONFIG.visual["translate:detect-language-override"]] == "off") {
+				return Utils.detectLanguage(this.state.currentLyrics);
+			}
+			return CONFIG.visual["translate:detect-language-override"];
+		};
+		console.log(language());
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:693]:
```suggestion
```

File: CustomApps/lyrics-plus/index.js [hunk 3]
```diff
@@ -339,13 +347,27 @@ class LyricsContainer extends react.Component {
...
+		console.log(lyricsState);
+		if (!lyricsState) return undefined;
+		this.state.currentLyrics = lyricsState;
+		switch (CONFIG.visual["translate:translated-lyrics-source"]) {
+			case "neteaseTranslation": {
+				if (this.state.neteaseTranslation !== null) this.state.currentLyrics = this.state.neteaseTranslation;
+				break;
+			}
+		}
+		console.log(this.state.currentLyrics);
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:361]:
```suggestion
```

File: CustomApps/lyrics-plus/OptionsMenu.js [hunk 4]
```diff
@@ -167,6 +190,8 @@ const TranslationMenu = react.memo(({ showTranslationButton, friendlyLanguage, h
 								lyricContainerUpdate && lyricContainerUpdate();
 								CONFIG.visual[name] && Spicetify.showNotification("Translating...", false, 5000);
 								translator.injectExternals();
+								// run the translate function after user changes translate lyrics source
+								// update the dropdowns
```
Comment (rxri) [inline CustomApps/lyrics-plus/OptionsMenu.js:194]:
```suggestion
```

**Review by rxri — DISMISSED** [review]

**Comment by Deskehs** [comment]
I removed the debug messages. The only thing left to do is to update the dropdowns after either a translation source/detect language override is selected, however, I'm not really that good with React and since the menu is memoized it's impossible to update. For my skills available it's hard to fix this.

**Comment by Deskehs** [comment]
I'll just submit it for review and let someone fix that dropdown thing. If anyone wants to take up on that I'll be glad.

**Comment by Deskehs** [comment]
hold up found some bugs

**Comment by Deskehs** [comment]
this should be fixed now

File: CustomApps/lyrics-plus/index.js [hunk 5]
```diff
@@ -339,17 +347,34 @@ class LyricsContainer extends react.Component {
...
 	}
 
+	lyricsSource(mode) {
+		const lyricsState = this.state[getKeyByValue(Modes, mode).toLowerCase()];
+		if (!lyricsState) return;
+		if (this.state[CONFIG.visual["translate:translated-lyrics-source"]]) {
+			this.state.currentLyrics = this.state[CONFIG.visual["translate:translated-lyrics-source"]];
+		} else {
+			this.state.currentLyrics = lyricsState;
+		}
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:357]:
```suggestion
		this.state.currentLyrics = this.state[CONFIG.visual["translate:translated-lyrics-source"]] ?? lyricsState
```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by rxri — APPROVED** [review]

File: CustomApps/lyrics-plus/index.js [hunk 6]
```diff
@@ -29,10 +29,14 @@ function getConfig(name, defaultVal = true) {
...
-	UNSYNCED = 2,
-	GENIUS = 3;
+const Modes = {
+	KARAOKE: 0,
+	SYNCED: 1,
+	UNSYNCED: 2,
+	GENIUS: 3
+};
+
+const getKeyByValue = (obj, value) => Object.keys(obj).find(key => obj[key] === value);
```
Comment (kyrie25) [inline CustomApps/lyrics-plus/index.js:39]: Object only serves the purpose of the function below, which was only used once and is not needed. This also adds a lot of boilerplate code.
Revert all unnecessary changes following this.

**Review by kyrie25 — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/index.js (hunk 6 above)
Comment (Deskehs) [reply CustomApps/lyrics-plus/index.js:39]: i will finish this up probably today

**Review by kyrie25 — APPROVED** [review]
